src/main/webapp/app/shared/metis/postings-reactions-bar/post-reactions-bar/post-reactions-bar.component.ts

```diff
@@ -19,4 +40,46 @@ export class PostReactionsBarComponent extends PostingsReactionsBarDirective<Pos
         reaction.post = this.posting;
         return reaction;
     }
+
+    /*
+    flips the value of the pinned property on a posting by invoking the metis service
+     */
+    togglePin() {
+        this.metisService.updatePostPinState(this.posting, !this.posting.pinned).subscribe(() => {});
+    }
+
+    /*
+    flips the value of the archived property on a posting by invoking the metis service
+     */
+    toggleArchive() {
+        this.metisService.updatePostArchiveState(this.posting, !this.posting.archived).subscribe(() => {});
+    }
+
+    /*
+    provides the tooltip for the pin icon dependent on the user authority and the pin state of a posting
+     */
+    getPinTooltip(): string {
+        if (this.currentUserIsAtLeastTutor && this.posting.pinned) {
+            return 'artemisApp.metis.removePinPostTutorTooltip';
+        }
+        if (this.currentUserIsAtLeastTutor && !this.posting.pinned) {
+            return 'artemisApp.metis.pinPostTutorTooltip';
+        } else {
+            return 'artemisApp.metis.pinnedPostTooltip';
+        }
```

Comment:
I think we could do just here and below:
```suggestion
        }
        return 'artemisApp.metis.pinnedPostTooltip';
```
The if-if-else looked a bit strange at first (expecting else-if's or similar).

Reply:
I will change it.